### PR TITLE
Feature/489169 projects descriptions acquisition

### PR DESCRIPTION
**Pull request by otrnovec**
a script that takes the project URL from the .csv file with the projects' information and extracts project descriptions and public interests from the XML. Those are then saved to a .csv with the project ID.

### CODE REVIEW

File: extract_project_descriptions_and_public_interests.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    output_df = pd.DataFrame(
+        {"project_id": list_of_project_ids,
+        "project_description": list_of_project_descriptions,
+        "public_interest": list_of_public_interest
+        })
+    
+    #save the dataframe to a csv file
+    output_df.to_csv('descriptions_and_interests.csv', sep=',', encoding='utf-8')
+
+
```
Comment (janrygl) [inline extract_project_descriptions_and_public_interests.py:79]: Please wrap this to 
if __name__ == "__main__"
construct, so it is not performed when somebody imports your functions

File: extract_project_descriptions_and_public_interests.py [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+#convert csv to df
+df = pd.read_csv("data\ProjektyPARO_5358953113614861487.csv")
```
Comment (janrygl) [inline extract_project_descriptions_and_public_interests.py]: make a file where you setup project dir,
e.g. settings.py in root directory having
PROJECT_PATH = Path(__file__).parent
end then setup other dirs:
DATA_PATH = PROJECT_PATH / "data"
and reuse them

File: extract_project_descriptions_and_public_interests.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+#extract the url list
+url_list = df["properties.detail"]
+
+#call the extract function
+lists = extract_project_description_and_public_interest(url_list)
```
Comment (janrygl) [inline extract_project_descriptions_and_public_interests.py]: lists is not good name, lists of what?

File: extract_project_descriptions_and_public_interests.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+#call the save function
```
Comment (janrygl) [inline extract_project_descriptions_and_public_interests.py]: everybody can guess that `save_to_csv` is calling a save function to csv
your method name says more than comment => remove such comments

File: extract_project_descriptions_and_public_interests.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    """Function to save the lists to a csv file."""
+    #convert lists to dataframe
+    output_df = pd.DataFrame(
+        {"project_id": list_of_project_ids,
+        "project_description": list_of_project_descriptions,
+        "public_interest": list_of_public_interest
+        })
+    
+    #save the dataframe to a csv file
+    output_df.to_csv('descriptions_and_interests.csv', sep=',', encoding='utf-8')
```
Comment (janrygl) [inline extract_project_descriptions_and_public_interests.py]: output should be parameter and not fixed path of function

File: extract_project_descriptions_and_public_interests.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
+""" This file is for getting the project descriptions and 
+public interests from xml and save them to a csv file. """
+import time
+import re
+import requests
+from bs4 import BeautifulSoup
```
Comment (janrygl) [inline extract_project_descriptions_and_public_interests.py:9]: It is a good practice to separate native imports from installed imports by empty line

**Review by janrygl — APPROVED** [review]
Nice, readable, a bit over-commented.:-)

File: extract_project_descriptions_and_public_interests.py [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+import pandas as pd
+
+
+def extract_project_description_and_public_interest(url_df):
+    """Function to extract project description and public interest from the xml."""
+
+    #initialize lists
```
Comment (janmalyy) [inline extract_project_descriptions_and_public_interests.py]: would be better to have a space after #

File: extract_project_descriptions_and_public_interests.py [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+public interests from xml and save them to a csv file. """
+import time
+import re
+import requests
+from bs4 import BeautifulSoup
+import pandas as pd
+
+
+def extract_project_description_and_public_interest(url_df):
+    """Function to extract project description and public interest from the xml."""
```
Comment (janmalyy) [inline extract_project_descriptions_and_public_interests.py]: the description of parameters is missing (but it's not that big problem as the names of variables are understandable)

Comment (janmalyy) [reply extract_project_descriptions_and_public_interests.py]: same for other functions

File: extract_project_descriptions_and_public_interests.py (hunk 5 above)
Comment (janmalyy) [inline extract_project_descriptions_and_public_interests.py]: why do you use comma as a separator and not semicolon?

File: data/ProjektyPARO_5358953113614861487.csv
Comment (janmalyy) [inline data/ProjektyPARO_5358953113614861487.csv:1]: there is some remaining noise in the data
<<<<<<< HEAD before and ======= after each sample

File: extract_project_descriptions_and_public_interests.py (hunk 1 above)
Comment (janmalyy) [inline extract_project_descriptions_and_public_interests.py:79]: I would appreciate more visible line between code execution and the functions

**Review by janmalyy — COMMENTED** [review]
Overall it is very nice, the code is clean, there are comments, its understandable.

File: extract_project_descriptions_and_public_interests.py (hunk 5 above)
Comment (janrygl) [reply extract_project_descriptions_and_public_interests.py]: actually,
csv = comma-separated value
tsv = tab-separated value

But people are using semicolons, tabs and commas for csv. Comma is fine, I personally like tabs better when I want to view data from console.

File: extract_project_descriptions_and_public_interests.py (hunk 8 above)
Comment (janrygl) [reply extract_project_descriptions_and_public_interests.py]: arg description is not must if everything is clear

but `Function to ` is boilerplate, better
`"""Extract project description and public interest from the DESCRIBE WHICH xml."""`

also, type annotations can be nice in modern Python

**Pull request merged**